dll/win32/shell32/CDefView.cpp

```diff
@@ -723,6 +723,17 @@ void CDefView::UpdateStatusbar()
         m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETICON, 2, pIcon, &lResult);
         m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 2, (LPARAM)szPartText, &lResult);
     }
+
+    SFGAOF att = 0;
+    if (cSelectedItems > 0)
+    {
+        UINT maxquery = 42; // Don't waste too much time here (_DoCopyToMoveToFolder will verify the full array)
```

Comment:
What do you mean by "don't waste too much time here"? In what sense?

Reply:
In the sense that `GetSelections` will build a PIDL array and then `GetAttributesOf` will inspect every pidl and the selection change code is already very slow because of the size calculation and I don't want to make it even slower so I simply don't do it if there are "too many" items. I don't think any folder is going to have more that 40 regitems in it so most likely we will inspect a few filesystem items as well and this sniffing should be adequate.

Ideally we would cache the attributes of the selection (there is a comment saying we should cache IIRC). `IShellItemArray::GetAttributes` (not implemented yet) is documented to cache the attributes, hopefully we can come up with some common code for that.

Reply:
`GetSelections` does not check the attributes, it just builds the array of PIDLs (two `SendMessage`s per item). DefView should probably get a function called something like `GetSelectionAttributes` and that function would cache.